tests/test_lab_generate.py

```diff
@@ -162,6 +162,34 @@ def test_OpenAI_server_error(self, get_instructions_from_model):
                 get_instructions_from_model.assert_called_once()
                 mt.teardown()
 
+    def test_new_data_too_long(self):
+        runner = CliRunner()
+        with open("tests/testdata/skill_too_long_answer.yaml", "rb") as qnafile:
+            with runner.isolated_filesystem():
+                mt = MockTaxonomy(pathlib.Path("taxonomy"))
+                mt.create_untracked(
+                    "compositional_skills/tracked/qna.yaml", qnafile.read()
+                )
+                result = runner.invoke(
+                    lab.generate,
```

Comment:
Where do you set the small context size?

Reply:
I do not lower the context size there, but instead use a yaml file that is long enough to cause this problem - that's the added `tests/testdata/skill_too_long_answer.yaml` file. However, this does rely on the default context length of lab generate not changing from the 4096 it has today, so I'll figure out how to amend this to explicitly set a context length on the server-side in this test to ensure even if the default changes from 4096 the test still does a proper context length check explicitly.

Reply:
Ohhh ok, thanks for clarifying. Right, amending by forcing `max_ctx_size` might make the test more resilient. If you want to address it soon I'll maintain my request change, but I don't see this as a blocker so if you want to do this as a followup I'm fine with that and I'll approve it. Just let me know.

Reply:
I'll go ahead and fix this as well, since I have to make other changes in the test after rebasing on top of the last month's worth of commits.

Reply:
ca5a9666b59324e0abe9ead65af0523e87a114ab sets a small context size explicitly, so marking this comment thread as resolved.